src/session.ts

```diff
@@ -712,4 +723,46 @@ function addTrack(this: WebPhoneSession, remoteAudioEle, localAudioEle): void {
     localAudio.play().catch(() => {
         this.logger.log('Local play was rejected');
     });
+    if (localStream && remoteStream && !this.mediaStatsStarted) {
+        this.mediaStreams = new MediaStreams(this);
+        this.logger.log('Start gathering media report');
+        this.mediaStatsStarted = true;
+        this.mediaStreams.getMediaStats((report: RTPReport) => {
+            this.logger.log(`Got media report: ${JSON.stringify(report)}`);
+            if (!this.reinviteForNoAudioSent && isNoAudio(report)) {
+                this.logger.log('No audio report');
+                this.noAudioReportCount++;
+                if (this.noAudioReportCount === 3) {
+                    this.logger.log('No audio for 6 sec. Try to recover audio by send Reinvite');
+                    this.mediaStreams.reconnectMedia();
+                    this.reinviteForNoAudioSent = true;
+                    this.noAudioReportCount = 0;
+                }
+            } else if (!isNoAudio(report)) {
+                this.noAudioReportCount = 0;
+            }
+        }, 2000)
+    }
+}
+
+function isNoAudio(report: RTPReport): boolean {
+    if (!report.inboundRtpReport) {
+        return true;
+    }
+    if (!report.outboundRtpReport) {
+        return true;
+    }
+    if (report.inboundRtpReport.packetsReceived === 0 || report.outboundRtpReport.packetsSent === 0) {
+        return true;
+    }
+    return false;
+}
```

Comment:
This should reside in `src/rtpReport.ts`.

Reply:
moved into rtpReport.ts